pgmpy/inference/ApproxInference.py

```diff
@@ -165,6 +177,34 @@ def query(
                 if virtual_evidence is None:
                     virtual_evidence = dict()
 
+                # Validate time slices in evidence
+                for var, state in evidence.items():
+                    if not isinstance(var, tuple) or len(var) != 2:
+                        raise ValueError(
+                            f"Invalid variable format in evidence: {var}. Expected (node, time_slice)."
+                        )
+                    if var[1] < 0:
+                        raise ValueError(
+                            f"Invalid time slice in evidence: {var[1]}. Time slice must be non-negative."
+                        )
+                    if var not in self.model.states:
+                        raise KeyError(f"Variable {var} not found in model states.")
+                # Validate virtual evidence normalization and time slices
+                for cpd in virtual_evidence:
+                    values = (
```

Comment:
Why do we need to convert it to numpy? We have functions in compat_fns.py that work on both numpy and torch array/tensor.